File: BFS/Java/Node.java
package BFS.Java;

/**
 * Node class for the BFS implementation.
 * Each node contains a value and references to its neighbors.
 */
public class Node {
    private int value;
    private Node[] neighbors;

    /**
     * Constructor to create a new node with the given value.
     * 
     * @param value The value to be stored in the node
     */
    public Node(int value) {
        this.value = value;
        this.neighbors = new Node[0];
    }

    /**
     * Get the value stored in this node.
     * 
     * @return The value stored in this node
     */
    public int getValue() {
        return value;
    }

    /**
     * Set the value for this node.
     * 
     * @param value The new value to be stored
     */
    public void setValue(int value) {
        this.value = value;
    }

    /**
     * Get the neighbors of this node.
     * 
     * @return Array of neighbor nodes
     */
    public Node[] getNeighbors() {
        return neighbors;
    }

    /**
     * Set the neighbors of this node.
     * 
     * @param neighbors Array of neighbor nodes
     */
    public void setNeighbors(Node[] neighbors) {
        this.neighbors = neighbors;
    }

    /**
     * Add a neighbor to this node.
     * 
     * @param neighbor The neighbor node to add
     */
    public void addNeighbor(Node neighbor) {
        Node[] newNeighbors = new Node[neighbors.length + 1];
        System.arraycopy(neighbors, 0, newNeighbors, 0, neighbors.length);
        newNeighbors[neighbors.length] = neighbor;
        this.neighbors = newNeighbors;
    }
}

File: BFS/Python/node.py
"""
Node class for BFS implementation.
This class represents a node in a graph or tree structure.
"""

class Node:
    def __init__(self, value):
        """
        Constructor to create a new node.
        
        Args:
            value: The value to be stored in the node
        """
        self.value = value
        self.neighbors = []  # List of neighboring nodes
        self.visited = False  # Flag to track if node has been visited during traversal
    
    def get_value(self):
        """
        Get the value stored in the node.
        
        Returns:
            The value stored in the node
        """
        return self.value
    
    def set_value(self, value):
        """
        Set the value stored in the node.
        
        Args:
            value: The new value to be stored
        """
        self.value = value
    
    def get_neighbors(self):
        """
        Get the list of neighboring nodes.
        
        Returns:
            List of neighboring nodes
        """
        return self.neighbors
    
    def add_neighbor(self, neighbor):
        """
        Add a neighbor to this node.
        
        Args:
            neighbor: The node to be added as a neighbor
        """
        if neighbor not in self.neighbors:
            self.neighbors.append(neighbor)
    
    def remove_neighbor(self, neighbor):
        """
        Remove a neighbor from this node.
        
        Args:
            neighbor: The node to be removed from neighbors
        """
        if neighbor in self.neighbors:
            self.neighbors.remove(neighbor)
    
    def is_visited(self):
        """
        Check if the node has been visited.
        
        Returns:
            True if the node has been visited, False otherwise
        """
        return self.visited
    
    def set_visited(self, visited):
        """
        Set the visited status of the node.
        
        Args:
            visited: Boolean indicating whether the node has been visited
        """
        self.visited = visited
    
    def reset_visited(self):
        """
        Reset the visited status to False.
        """
        self.visited = False
